src/ripple/shamap/SHAMapTreeNode.h

```diff
@@ -39,7 +39,7 @@ enum SHANodeFormat
     snfHASH     = 3, // just the hash
 };
 
-class SHAMapTreeNode
+class SHAMapAbstractNode
```

Comment:
Suggested future change: Since you renamed and split the class, rename and split the file, too. Though I do appreciate you didn't do it in this PR. :)